Fix Testname missing in conformance.yaml

Jefftree opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

Fix Testname missing in conformance.yaml https://github.com/kubernetes/kubernetes/issues/88921

Some conformance tests had their header omit Testname and replaced with Name instead

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

**Special notes for your reviewer**:

This PR rebases off of https://github.com/kubernetes/kubernetes/pull/88770

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

/sig testing
/area conformance
/assign @johnbelamaric

Jefftree committed:
Include testname

Jefftree commented:
/assign @johnbelamaric @yujuhong
/retest

Jefftree commented:
/cc @oomichi

dims commented:
/milestone v1.18
/priority important-soon

This is clearly a bug.

dims commented:
@Jefftree do we throw an error when `Testname` is missing? Can we please add that if not?

dims commented:
This validation will become important when we start scripting using https://github.com/kubernetes/kubernetes/pull/88924

Jefftree committed:
Only check testname for missing

Jefftree commented:
@dims The behavior before was that we only throw an error if BOTH Testname and Release fields were empty, I changed it to just look at Testname for now.

@johnbelamaric We have some conformance tests without the Release tag, and the validation did not catch those. I removed the check for the Release tag as a temporary measure, but we should look into whether those need to be backfilled or whether it's safe to ignore Release

Jefftree commented:
/test pull-kubernetes-bazel-test

dims commented:
/assign @johnbelamaric @yujuhong @liggitt

dims commented:
/assign @spiffxp

cpanato commented:
ping @liggitt @johnbelamaric to review and merge if all agree, we need to get that in to cut the RC today

johnbelamaric commented:
I didn't realize this was targeted at v1.18. @dims @Jefftree why do we need it in 1.18? While I believe there is no risk because it is just a tooling change, I prefer not to merge anything in 1.18 at this point that's not critical.

johnbelamaric commented:
/milestone v1.19

johnbelamaric commented:
/lgtm
/approve

dims commented:
@liggitt i don't know if folks listed in `test/e2e/common/OWNERS` will be interested in this. So please take a look when you get a chance.

oomichi commented on test/e2e/common/runtime.go:
```diff
@@ -189,7 +189,7 @@ while true; do sleep 1; done
 			/*
 				Release: v1.15
-				Name: Container Runtime, TerminationMessagePath, non-root user and non-default path
+				Testname: Container Runtime, TerminationMessagePath, non-root user and non-default path
```
Confirmed this 4 parts are last ones what `Name:` instead of `Testname:` under test/e2e with `$ grep "\sName: " * -R | grep -v '"'`

oomichi reviewed (commented):
/lgtm
/approve
